internal/check/handler.go

```diff
@@ -106,6 +120,20 @@ func (h *Handler) getCheck(w http.ResponseWriter, r *http.Request, _ httprouter.
 	h.d.Writer().WriteCode(w, r, http.StatusForbidden, &RESTResponse{Allowed: false})
 }
 
+func (h *Handler) ensureDepthQueryParam(w http.ResponseWriter, r *http.Request) (int, bool)  {
```

Comment:
This helper function could also be used by the expand API right? We could refactor it to be in the `internal/x` package for example.

Reply:
I would propose to then make it `func GetMaxDepthFromQuery(q *url.Values) error`.

Reply:
Thanks. Addressed